connect/src/spirc.rs

```diff
@@ -785,13 +791,29 @@ impl SpircTask {
             return Err(SpircError::Ident(ident.to_string()).into());
         }
 
+        let old_client_id = self.session.client_id();
+        let old_client_name = self.session.client_name();
+
         for entry in update.get_device_state().get_metadata().iter() {
             if entry.get_field_type() == "client_id" {
                 self.session.set_client_id(entry.get_metadata());
                 break;
             }
         }
 
+        self.session
+            .set_client_name(update.get_device_state().get_name());
+
+        let new_client_id = self.session.client_id();
+        let new_client_name = self.session.client_name();
+
+        if self.device.get_is_active()
+            && (new_client_id != old_client_id || new_client_name != old_client_name)
+        {
+            self.player
+                .emit_session_client_changed_event(new_client_id, self.session.client_name());
```

Comment:
Probably not necessary, to call this method here again.

```suggestion
                .emit_session_client_changed_event(new_client_id, new_client_name);
```

Reply:
Thanks. (I know I keep saying that,lol!!!) I'll roll all of your suggestions into one commit now that I'm home.